sunpy/physics/differential_rotation.py

```diff
@@ -227,102 +411,138 @@ def _warp_sun_coordinates(xy, smap, dt: u.s, **diffrot_kwargs):
 
     # Re-stack the data to make it correct output form
     xy2 = np.dstack([x2.T.value.flat, y2.T.value.flat])[0]
+
     # Returned a masked array with the non-finite entries masked.
-    xy2 = np.ma.array(xy2, mask=np.isnan(xy2))
-    return xy2
+    return np.ma.array(xy2, mask=np.isnan(xy2))
 
 
-@u.quantity_input
-def diffrot_map(smap, time=None, dt: u.s=None, pad=False, **diffrot_kwargs):
+def differential_rotate(smap, observer=None, time=None, **diff_rot_kwargs):
     """
     Function to apply solar differential rotation to a sunpy map.
 
     Parameters
     ----------
-    smap : `~sunpy.map`
+    smap : `~sunpy.map.GenericMap`
         Original map that we want to transform.
+
+    observer : `~astropy.coordinates.BaseCoordinateFrame`, `~astropy.coordinates.SkyCoord`
+        The location of the new observer.
+        Instruments in Earth orbit can be approximated by using the position
+        of the Earth at the observation time of the new observer.
+
     time : sunpy-compatible time
```

Comment:
This might be too complex to explain how to do for this here. 
```suggestion
    time : sunpy-compatible time, optional
```

Reply:
yes, unfortunately it is a bit complicated.  I've put a description of the logic in the _get_new_observer helper function, since that is where the logic is implemented.